### PR TITLE
Change treatment of generic column type `object`

### PR DESCRIPTION
**Pull request by Louquinze**
This PR changes the treatment of columns with dtype `object`. This columns will be treated as `string`.

### CODE REVIEW

File: autosklearn/data/feature_validator.py [hunk 1]
```diff
@@ -327,7 +325,7 @@ def get_feat_type_from_columns(
                 else:
                     raise ValueError(
                         "Input Column {} has unsupported dtype {}. "
-                        "Supported column types are categorical/bool/numerical dtypes. "
+                        "Supported column types are categorical/bool/numerical/string dtypes. "  # noqa: E501
```
Comment (mfeurer) [inline autosklearn/data/feature_validator.py:328]: I don't think this is necessary. Please reformat the string so the lines fit within the line limit.

File: autosklearn/data/feature_validator.py [hunk 2]
```diff
@@ -304,16 +305,13 @@ def get_feat_type_from_columns(
...
-                        "pandas.Series.astype ."
-                        "If working with string objects, the following "
-                        "tutorial illustrates how to work with text data: "
-                        "https://scikit-learn.org/stable/tutorial/text_analytics/working_with_text_data.html"  # noqa: E501
+                    warnings.warn(
+                        f"Input Column {column} has generic type object. "
+                        f"Autosklearn will treat this column as string. "
+                        f"Please ensure that this setting is suitable for your task."
                     )
+                    X[column] = X[column].astype("string")
```
Comment (mfeurer) [inline autosklearn/data/feature_validator.py:313]: Does this work for random objects? Could we have a test that the feature validator correctly handles random objects? In general, could you please extend the tests under `test/test_data/test_feature_validator.py`?

**Review by mfeurer — DISMISSED** [review]

File: autosklearn/data/feature_validator.py (hunk 1 above)
Comment (Louquinze) [reply autosklearn/data/feature_validator.py:328]: i change it

File: autosklearn/data/feature_validator.py (hunk 2 above)
Comment (Louquinze) [reply autosklearn/data/feature_validator.py:313]:
```python
class Dummy:
    def __init__(self, x):
        self.x = x
    def __call__(self):
        print(self.x)
    def dummy_func(self):
        for i in range(100):
            print("do something 100 times")
            
dummy = Dummy(1)
dummy_2 = Dummy(2)
dummy_3 = Dummy(3)
df = pd.DataFrame({"Test object": [dummy], "Test list of objects": [[dummy_2, dummy_3]]})
df.info()
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 2 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Test object           1 non-null      object
 1   Test list of objects  1 non-null      object
dtypes: object(2)
memory usage: 144.0+ bytes
df = df.astype({"Test object": "string", "Test list of objects": "string"})
df.info()
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1 entries, 0 to 0
Data columns (total 2 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Test object           1 non-null      string
 1   Test list of objects  1 non-null      string
dtypes: string(2)
memory usage: 144.0 bytes
print(df)
                                 Test object                               Test list of objects
0  <__main__.Dummy object at 0x7fcfbcf880a0>  [<__main__.Dummy object at 0x7fcff7cb2ac0>, <_...

```

**Comment by Louquinze** [comment]
> Looks good to me, include a docstring if you like, otherwise I'll merge :)

just merge, i will add the docstring in a new PR

**Pull request merged**